tests/stdlib/tlists.nim

```diff
@@ -146,5 +146,62 @@ template testCommon(initList, toList) =
           ci
       doAssert s == [0, 1, 0, 1, 0, 1]
 
+  block: # prepend, prependMoved
+    block:
+      var
+        l0 = initList[int]()
+        l1 = [1].toList
+        l2 = [2, 3].toList
+        l3 = [4, 5, 6].toList
+      l0.prepend l3
+      l1.prepend l3
+      doAssert l3.toSeq == [4, 5, 6]
+      l2.prependMoved l3
+      doAssert l0.toSeq == [4, 5, 6]
+      doAssert l1.toSeq == [4, 5, 6, 1]
+      doAssert l2.toSeq == [4, 5, 6, 2, 3]
+      doAssert l3.toSeq == []
+      l2.prepend l3 # re-prepending l3 that was destroyed is now a no-op
+      doAssert l2.toSeq == [4, 5, 6, 2, 3]
+      doAssert l3.toSeq == []
+    block:
+      var
+        l0 = initList[int]()
+        l1 = [1].toList
+        l2 = [2, 3].toList
+        l3 = [4, 5, 6].toList
+      l3.prependMoved l0
+      l2.prependMoved l1
+      doAssert l3.toSeq == [4, 5, 6]
+      doAssert l2.toSeq == [1, 2, 3]
+      l3.prepend l0
+      doAssert l3.toSeq == [4, 5, 6]
+    block:
+      var c = [0, 1].toList
+      c.prependMoved c
+      let s = collect:
+        for i, ci in enumerate(c):
+          if i == 6: break
+          ci
+      doAssert s == [0, 1, 0, 1, 0, 1]
+
+  block remove:
+    var l = [0, 1, 2, 3].toList
```

Comment:
* also add this test case where you remove an element from a cycle:
```nim
when true:
  import lists
  template fn(init) =
    var a = [10,11,12].toDoublyLinkedList
    a.addMoved a
    doAssert take(a, 6) == @[10, 11, 12, 10, 11, 12]
    remove(a, a.head.next)
    doAssert take(a, 6) == @[10, 12, 10, 12, 10, 12]
  fn(toDoublyLinkedList)
  fn(toSinglyLinkedList)
```

* and specify in doc comment for both `remove` overloads that removing an element from a cycle maintains a cycle (it's not obvious, it could've also cut the cycle; but note that for for `DoublyLinkedList`, since we're limited to O(1) for remove, that's the only possible behavior, so might as well preserve this behavior for `SinglyLinkedList`)
* and also specify this behavior in a runnableExamples (can combine into 1 to avoid duplication)

* I'm using this definition for take, which I want to add to sequtils:
```nim
  template take(a: untyped, max: int): untyped =
    type T = typeof(block: (for ai in a: ai))
    var ret: seq[T]
    var i = 0
    if max > 0:
      for ai in a:
        ret.add ai
        i.inc
        if i >= max: break
    ret
```
you can use this in tslists.nim, but for runnableExamples, you can use the usual enumerate

Reply:
At this point I am starting to doubt the usefulness of the ...Ring types. Maybe they should be removed, and replaced with just this:
```nim
proc toRing[T](a: var SomeLinkedList) = a.addMoved a
```

Reply:
How can I combine the runnableExamples of two different `remove` procs into one?

Reply:
just refer to it, eg:
```nim
see examples for <link...>
```

Reply:
> At this point I am starting to doubt the usefulness of the ...Ring types. Maybe they should be removed, and replaced with just this:

semantics are different, eg `toSeq(someRing)` makes sense, whereas `toSeq(someLinkedListWithLoop)` will go in infinite loop and then likely go out of memory